packages/components/src/tools-panel/tools-panel-item/hook.ts

```diff
@@ -108,9 +104,20 @@ export function useToolsPanelItem(
 		? menuItems?.[ menuGroup ]?.[ label ] !== undefined
 		: isMenuItemChecked;
 
+	const placeholderStyle =
+		shouldRenderPlaceholder &&
+		! isShown &&
+		styles.ToolsPanelItemPlaceholder;
+
+	const cx = useCx();
+	const classes = useMemo( () => {
+		return cx( styles.ToolsPanelItem, placeholderStyle, className );
+	}, [ isShown, shouldRenderPlaceholder, className ] );
+
```

Comment:
Let's move the whole classname computation inside `useMemo`

```suggestion
	const cx = useCx();
	const classes = useMemo( () => {
		const placeholderStyle =
			shouldRenderPlaceholder &&
			! isShown &&
			styles.ToolsPanelItemPlaceholder;
		return cx( styles.ToolsPanelItem, placeholderStyle, className );
	}, [ isShown, shouldRenderPlaceholder, className ] );
```